### DISCUSSION TITLE
How to get classification results/labels from survival data

### DISCUSSION THREAD

**Discussion by jianxinwang** [kogalur/randomForestSRC#369] [Q&A]
I have a training dataset that contains survival time and survival status for some patients. After training a survival forest, I would like to obtain prediction results on a testing dataset with labels like high- or low-risk group. With this classification, I would like to stratify patients with this label and then plot a K-M plot. How to extract this information? Or I can use a quantitative risk score to stratify the patients. What score to use and how to get it? Sorry, I'm new to using this package.

**Comment by ishwaran** [comment]
Here's an illustration for the PBC data set.  In this example, patients are stratified by mortality (see the help file/vignettes/getting started for the definition of "mortality") into quartiles.  Then we plot the averaged ensemble survival for each group against time.

```
## run rsf on pbc data
data(pbc)
o <- rfsrc(Surv(days, status) ~ ., pbc)

## group the data using quartiles for OOB mortality
yhat <- o$predicted.oob
q.yhat <- quantile(yhat)
q.yhat[1] <- q.yhat[1] - .0001
grp <- cut(yhat, q.yhat, labels = FALSE)

## mean OOB survival for each OOB mortality quartile
surv.gp <- 100 * do.call(cbind, lapply(unique(grp), function(gr) {
  colMeans(o$survival.oob[grp==gr,, drop=FALSE], na.rm = TRUE)
}))

## extract the event times
tme <- o$time.interest

## plot survival by event time, per group
matplot(tme, surv.gp, type ="l", lty=1,
        xlab="time",
        ylab="survival stratifed by OOB mortality")

```

**Reply by jianxinwang** [reply]
Thanks for the quick replay. I followed your instructions and created risk groups (values on the low end of yhat correspond to shorter predicted survival time and higher values correspond to longer predicted survival time). When I make a Kaplan-Meier plot based on this grouping, the longer survival group corresponds to group 1 and the shorter survival group corresponds to group 2. I think it should be the other way around. Did I misunderstand something?

**Reply by randomForestSRC** [reply]
The predicted value “yhat” is mortality which estimates expected deaths. Higher values correspond to shorter life expectancy. So it’s the inverse of what you said.

**Comment by jianxinwang** [comment]
Great, thank you!

**Reply by jianxinwang** [reply]
Here is my other question: my training dataset has max survival time of 80 months, however, my test dataset has max survival time of 300 months. Can I used the trained model to predict survival of my test dataset?

**Reply by ishwaran** [reply]
Yes you can do that.  The package will provide an estimated survival function for each test data point regardless of the length of follow up.  The survival function is stored in the object `$survival`

```
#Here's an example from the help file for `predict.rfsrc`

## load veteran data and grow RSF on training data
data(veteran, package = "randomForestSRC")
train <- sample(1:nrow(veteran), round(nrow(veteran) * 0.80))
veteran.grow <- rfsrc(Surv(time, status) ~ ., veteran[train, ], ntree = 100) 

## use trained forest on test data
veteran.pred <- predict(veteran.grow, veteran[-train, ])

## for example, here's the estimated survival function for the first test data point
veteran.pred$survival[1,]
```

I want to point out that we have a [dedicated webpage](https://www.randomforestsrc.org) for the package with detailed vignettes, a getting started section and link to the help files for all functions, that you might find helpful.

**Reply by jianxinwang** [reply]
The trained model have never seen training data with survival times > 80 months. It is still able to predict a test dataset that contains many patients with survival time far exceeds 80 months (in my case, ~300 months). Because the prediction is based on a survival function, is that right? Yes, I'm also reading your documentations. That is very helpful.
